fix(flatten): fixed flatten not handling empty arrays

dhershman1 opened the pull request:
Signed-off-by: Dustin Hershman <dustinh17@gmail.com>

### :pencil2: Change Description

This should address the issue with flatten not properly handling empty arrays in an object, I didn't want to get to fancy with it so I left the solution pretty basic

### :link: Relates to Issue

Fixes #39 

### :triangular_flag_on_post: Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### :heavy_check_mark: Pull Request Checklist

- [x] Adherence to the  [Developer Certificate of Origin (DCO)](https://developercertificate.org/) version 1.1 (`git --signoff`).
- [x] Commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard.
- [x] Ensure all tests are passing (`npm test`).
- [x] Adhere to [standardjs](http://standardjs.com) guidelines.
- [x] Add tests for any new features or to show that bugs have been fixed.
- [x] Update the README.md with details of changes to the interface.

dhershman1 committed:
fix(flatten): fixed flatten not handling empty arrays

fix #39

Signed-off-by: Dustin Hershman <dustinh17@gmail.com>

dhershman1 committed:
fix(flatten): added support for empty objects

Turns out empty objects were also being skipped, so this fix should remedy that

fix #39

Signed-off-by: Dustin Hershman <dustinh17@gmail.com>

jessie-codes commented on src/index.js:
```diff
@@ -11,6 +11,12 @@ const flatten = (obj, delimiter) => {
 
   const flat = (original, stack, prev) => {
+    if (!Object.values(original).length) {
+      result[prev] = original
```
This should verify if `prev` exist before using it as a key. Currently this results in the following behavior:

```javascript
console.log(flatten({}))
// { undefined: {} }
```

dhershman1 replied:
Good catch! Just added `prev` to the if logic this should address the issue on hand.

dhershman1 committed:
fix(flatten): fixes flatten borking a completely empty object

This should address flatten creating or using undefined keys for empty objects

Signed-off-by: Dustin Hershman <dustinh17@gmail.com>
